x-pack/plugins/ml/public/application/model_management/test_models/models/text_expansion/text_expansion_inference.ts

```diff
@@ -0,0 +1,177 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { i18n } from '@kbn/i18n';
+import { SUPPORTED_PYTORCH_TASKS } from '@kbn/ml-trained-models-utils';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { trainedModelsApiProvider } from '../../../../services/ml_api_service/trained_models';
+import { InferenceBase, INPUT_TYPE, type InferResponse } from '../inference_base';
+import { getTextExpansionOutputComponent } from './text_expansion_output';
+import { getTextExpansionInput } from './text_expansion_input';
+
+export interface TextExpansionPair {
+  token: string;
+  value: number;
+}
+
+export interface FormattedTextExpansionResponse {
+  text: string;
+  score: number;
+  originalTokenWeights: TextExpansionPair[];
+  adjustedTokenWeights: TextExpansionPair[];
+}
+
+export type TextExpansionResponse = InferResponse<
+  FormattedTextExpansionResponse,
+  estypes.MlInferTrainedModelResponse
+>;
+
+export class TextExpansionInference extends InferenceBase<TextExpansionResponse> {
+  protected inferenceType = SUPPORTED_PYTORCH_TASKS.TEXT_EXPANSION;
+  protected inferenceTypeLabel = i18n.translate(
+    'xpack.ml.trainedModels.testModelsFlyout.textExpansion.label',
+    { defaultMessage: 'Text expansion' }
+  );
+  protected info = [
+    i18n.translate('xpack.ml.trainedModels.testModelsFlyout.textExpansion.info', {
+      defaultMessage:
+        'Expand your search to include relevant terms in the results that are not present in the query.',
+    }),
+  ];
+
+  public queryText$ = new BehaviorSubject<string>('');
+  private queryResults: Record<string, number> = {};
+
+  constructor(
+    trainedModelsApi: ReturnType<typeof trainedModelsApiProvider>,
+    model: estypes.MlTrainedModelConfig,
+    inputType: INPUT_TYPE,
+    deploymentId: string
+  ) {
+    super(trainedModelsApi, model, inputType, deploymentId);
+
+    this.initialize(
+      [this.queryText$.pipe(map((questionText) => questionText !== ''))],
+      [this.queryText$]
+    );
+  }
+
+  protected async inferText() {
+    return this.runInfer<estypes.MlInferTrainedModelResponse>(
+      () => {},
+      (resp, inputText) => {
+        return {
+          response: parseResponse(
+            resp as unknown as MlInferTrainedModelResponse,
+            '',
+            this.queryResults
+          ),
+          rawResponse: resp,
+          inputText,
+        };
+      }
+    );
+  }
+
+  protected async inferIndex() {
+    const { docs } = await this.trainedModelsApi.trainedModelPipelineSimulate(this.getPipeline(), [
+      {
+        _source: {
+          text_field: this.getQueryText(),
+        },
+      },
+    ]);
+    this.queryResults = docs[0].doc?._source[this.inferenceType].predicted_value ?? {};
```

Comment:
is it possible that `docs[0]` is undefined?

Reply:
There's a chance, so I've added a check in https://github.com/elastic/kibana/pull/159150/commits/faefba5fff6817c3b79e296a6749e1ed37bed2ef